panther_hardware_interfaces/include/panther_hardware_interfaces/panther_system.hpp

```diff
@@ -0,0 +1,136 @@
+#include <hardware_interface/system_interface.hpp>
+#include <hardware_interface/types/hardware_interface_return_values.hpp>
+
+#include <panther_hardware_interfaces/gpio_driver.hpp>
+#include <panther_hardware_interfaces/motors_controller.hpp>
+#include <panther_hardware_interfaces/panther_system_ros_interface.hpp>
+#include <panther_hardware_interfaces/roboteq_error_filter.hpp>
+
+namespace panther_hardware_interfaces
+{
+using return_type = hardware_interface::return_type;
+using CallbackReturn = rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn;
+using StateInterface = hardware_interface::StateInterface;
+using CommandInterface = hardware_interface::CommandInterface;
+
+/**
+ * @brief Class that implements SystemInterface from ros2_control for Panther
+ */
+class PantherSystem : public hardware_interface::SystemInterface
+{
+public:
+  RCLCPP_SHARED_PTR_DEFINITIONS(PantherSystem)
+
+  CallbackReturn on_init(const hardware_interface::HardwareInfo & hardware_info) override;
+  CallbackReturn on_configure(const rclcpp_lifecycle::State & previous_state) override;
+  CallbackReturn on_cleanup(const rclcpp_lifecycle::State & previous_state) override;
+  CallbackReturn on_activate(const rclcpp_lifecycle::State & previous_state) override;
+  CallbackReturn on_deactivate(const rclcpp_lifecycle::State & previous_state) override;
+  CallbackReturn on_shutdown(const rclcpp_lifecycle::State & previous_state) override;
+  CallbackReturn on_error(const rclcpp_lifecycle::State & previous_state) override;
+
+  std::vector<StateInterface> export_state_interfaces() override;
+  std::vector<CommandInterface> export_command_interfaces() override;
+
+  return_type read(const rclcpp::Time & time, const rclcpp::Duration & period) override;
+  return_type write(const rclcpp::Time & time, const rclcpp::Duration & period) override;
+
+protected:
+  void CheckJointSize();
+  void SortJointNames();
+  void CheckJointNames();
+  void SetInitialValues();
+  void CheckInterfaces();
+  void ReadDrivetrainSettings();
+  void ReadCanOpenSettings();
+  void ReadInitializationActivationAttempts();
+  void ReadParametersAndCreateRoboteqErrorFilter();
+
+  void UpdateHwStates();
+  void UpdateDriverState();
+  void UpdateSystemFeedback();
+  void UpdateMsgErrors();
+
+  static constexpr size_t kJointsSize = 4;
+
+  // Currently only velocity command mode is supported - although Roboteq driver support position
+  // and torque mode, in 2.1 firmware both modes aren't really stable and safe.
+  // In torque mode sometimes after killing the software motor moves and it generally isn't well
+  // tuned. Position mode also isn't really stable (reacts abruptly to spikes). If updating the
+  // firmware to 2.1a will solve these issues, it may be worth to enable other modes.
+  double hw_commands_velocities_[kJointsSize];
+
+  double hw_states_positions_[kJointsSize];
+  double hw_states_velocities_[kJointsSize];
+  double hw_states_efforts_[kJointsSize];
```

Comment:
```suggestion
    std::array<double, kJointsSize> hw_commands_velocities_;
  
  std::array<double, kJointsSize> hw_states_positions_;
  std::array<double, kJointsSize> hw_states_velocities_;
  std::array<double, kJointsSize> hw_states_efforts_;
```
Just to make it overly modern C++